src/app/shared/doc-viewer/doc-viewer.ts

```diff
@@ -113,14 +113,23 @@ export class DocViewer implements OnDestroy {
 
     [...exampleElements].forEach((element: Element) => {
       const example = element.getAttribute(componentName);
+      const lines = element.getAttribute('lines');
+      const file = element.getAttribute('file');
       const portalHost = new DomPortalOutlet(
           element, this._componentFactoryResolver, this._appRef, this._injector);
       const examplePortal = new ComponentPortal(componentClass, this._viewContainerRef);
       const exampleViewer = portalHost.attach(examplePortal);
       if (example !== null) {
         const exampleViewerComponent = exampleViewer.instance as ExampleViewer;
         exampleViewerComponent.example = example;
-        exampleViewerComponent.view = 'collapsed';
+        if (file) {
+          exampleViewerComponent.view = 'compact';
```

Comment:
Isn't there some boolean attribute that tells us whether it should be compact or full?